### PR TITLE
Resolve directory to the real path before using as base for resolve

### LINKED ISSUE #6: Use the real path for resolving dependencies
I created a similar issue at the [resolve](https://github.com/substack/node-resolve/issues/130) repo.

Node uses the real path of modules (except when executed with `--preserve-symlinks`) to resolve dependencies. That is why [pnpm](https://github.com/pnpm/pnpm)'s [symlinked `node_modules`](https://github.com/pnpm/pnpm/blob/master/docs/symlinked-node-modules-structure.md) work fine with Node applications.

However, `resolve-from` does not use the real path of modules to resolve dependencies. As a consequence, toolings that use `resolve-from` break on projects that use pnpm. These are issues related to `resolve` but I checked them with `resolve-from` and they fail with it as well: https://github.com/pnpm/pnpm/issues/857, https://github.com/pnpm/pnpm/issues/801

Would you accept a pull request to change the behavior of `resolve-from` to work the way Node does? It won't change anything for projects that use npm/Yarn because they don't leverage symlinks but it would be a **huge** help for the pnpm community!

### PR DESCRIPTION
**Pull request by zkochan**
Close #6

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,7 +12,15 @@ const resolveFrom = (fromDir, moduleId, silent) => {
 		throw new TypeError(`Expected \`moduleId\` to be of type \`string\`, got \`${typeof moduleId}\``);
 	}
 
-	fromDir = path.resolve(fromDir);
+	try {
+		fromDir = resolveLinkTarget.sync(fromDir);
```
Comment (sindresorhus) [inline index.js]: Why not just `fs.realpathSync()`?

Comment (zkochan) [inline index.js]: Indeed. Updated

File: index.js [hunk 2]
```diff
@@ -11,7 +12,15 @@ const resolveFrom = (fromDir, moduleId, silent) => {
 		throw new TypeError(`Expected \`moduleId\` to be of type \`string\`, got \`${typeof moduleId}\``);
 	}
 
-	fromDir = path.resolve(fromDir);
+	try {
+		fromDir = fs.realpathSync(fromDir);
+	} catch (err) {
+		if (err.code !== 'EINVAL') {
```
Comment (sindresorhus) [inline index.js]: Not really sure why you're just throwing on `EINVAL`. Any error should be thrown.

You also have to correctly handle the `silent` argument. See line 32.

Comment (zkochan) [inline index.js]: you're right, `EINVAL` was needed only when used with the `resolve-link-target` lib. All should be thrown when using `fs.realpathSync`. Also added tests for silent.

Thanks for the quick review. If anything else, I'll fix tomorrow. morning.

**Comment by kevva** [comment]
LGTM :+1:

File: index.js (hunk 2 above)
Comment (zkochan) [inline index.js]: @sindresorhus, after thinking about this more, I am not so sure we have to throw any error. Right now it works if the `fromDir` does not exist in the filesystem. [resolve](https://github.com/substack/node-resolve) seems to behave the same. So maybe we just need to use `path.resolve` on `fromDir` and then ignore `ENOENT` errors of `realpathSync`. What do you think?

**Comment by sindresorhus** [comment]
Sorry, forgot about this. I'm gonna do a major bump just in case it affects something.

**Pull request merged**